packages/cli/src/processed-data-managers/processed-data-helper.ts

```diff
@@ -0,0 +1,322 @@
+import { createHash } from 'crypto';
+import { ApplicationError } from 'n8n-workflow';
+import type {
+	ICheckProcessedContextData,
+	IProcessedDataManager,
+	ICheckProcessedOptions,
+	ICheckProcessedOutput,
+	ProcessedDataContext,
+	ProcessedDataItemTypes,
+	ProcessedDataMode,
+} from 'n8n-workflow';
+import { Container } from 'typedi';
+
+import { ProcessedDataRepository } from '@/databases/repositories/processed-data.repository';
+import type { IProcessedDataEntries, IProcessedDataLatest } from '@/interfaces';
+
+export class ProcessedDataHelper implements IProcessedDataManager {
+	private static sortEntries(
+		items: ProcessedDataItemTypes[],
+		mode: ProcessedDataMode,
+	): ProcessedDataItemTypes[] {
+		return [...items].sort((a, b) => (ProcessedDataHelper.compareValues(mode, a, b) ? 1 : -1));
+	}
+
+	private static compareValues(
+		mode: ProcessedDataMode,
+		value1: ProcessedDataItemTypes,
+		value2: ProcessedDataItemTypes,
+	): boolean {
+		if (mode === 'latestIncrementalKey') {
+			const num1 = Number(value1);
+			const num2 = Number(value2);
+			if (!isNaN(num1) && !isNaN(num2)) {
+				return num1 > num2;
+			}
+			throw new ApplicationError(
+				'Invalid value. Only numbers are supported in mode "latestIncrementalKey"',
+			);
+		} else if (mode === 'latestDate') {
+			const date1 = new Date(value1 as string);
+			const date2 = new Date(value2 as string);
+
+			if (!isNaN(date1.getTime()) && !isNaN(date2.getTime())) {
+				return date1 > date2;
+			} else {
+				throw new ApplicationError(
+					'Invalid value. Only valid dates are supported in mode "latestDate"',
+				);
+			}
+		} else {
+			throw new ApplicationError(
+				"Invalid mode. Only 'latestIncrementalKey' and 'latestDate' are supported.",
+			);
+		}
+	}
+
+	private static createContext(
+		context: ProcessedDataContext,
+		contextData: ICheckProcessedContextData,
+	): string {
+		if (context === 'node') {
+			console.log('contextData.node', contextData, contextData.node);
+			if (!contextData.node) {
+				throw new ApplicationError(
+					"No node information has been provided and can so not use context 'node'",
+				);
+			}
+			// Use the node ID to make sure that the data can still be accessed and does not get deleted
+			// whenver the node gets renamed
+			return `n:${contextData.node.id}`;
+		}
+
+		return '';
+	}
+
+	private static createValueHash(value: ProcessedDataItemTypes): string {
+		return createHash('md5').update(value.toString()).digest('base64');
+	}
+
+	async checkProcessed(
```

Comment:
This seems to split the items into new and processed. Could we name the function as such then? E.g. `splitToNewAndAlreadyProcessed`?